test/controllers/HealthCheckController.test.ts

```diff
@@ -0,0 +1,35 @@
+import 'reflect-metadata'
+import { Application } from 'express'
+import * as request from 'supertest'
+import { createApplication } from '../ApplicationFactory';
+import { createSubstituteOf } from '../SubstituteFactory';
+
+import '../../src/controllers/HealthCheckController'
+import { RedisService } from '../../src/services/RedisService';
+
+describe('HealthCheckController', () => {
+    it('should return 200 with status when redis database is up', async () => {
+        const app = createApplication(container => {
```

Comment:
The intention is to prepare application with targeted route and it's dependencies where dependencies are injected into test instance of the application using DI. When mocked instances are needed `createSubstituteOf` helper factory method can be used to created instance of a substitute with predefined behaviour.